trellis/containers/transforms.hpp

```diff
@@ -0,0 +1,233 @@
+     *
+     * @return the Eigen affine transform
+     */
+    AffineTransform3D GetAffineRepresentation() const {
+      return AffineTransform3D(Eigen::Translation<double, 3>(translation.x, translation.y, translation.z) *
+                               Eigen::Quaterniond(rotation.w, rotation.x, rotation.y, rotation.z));
+    }
+
+    /**
+     * Inverse retrieve the inverse transform
+     *
+     * @return the inverse transform
+     */
+    RigidTransform Inverse() const { return Transforms::RigidTransform(GetAffineRepresentation().inverse()); }
+
+    bool operator==(const RigidTransform& other) const {
+      return this->translation.x == other.translation.x && this->translation.y == other.translation.y &&
+             this->translation.z == other.translation.z && this->rotation.w == other.rotation.w &&
+             this->rotation.x == other.rotation.x && this->rotation.y == other.rotation.y &&
+             this->rotation.z == other.rotation.z;
+    }
+
+    bool operator!=(const RigidTransform& other) const { return !(*this == other); }
+
+    Translation translation;
+    Rotation rotation;
+
+   private:
+    static Translation GetTranslationFromAffineTransform(const AffineTransform3D& transform) {
+      return Translation{transform.translation().x(), transform.translation().y(), transform.translation().z()};
+    }
+    static Rotation GetRotationFromAffineTransform(const AffineTransform3D& transform) {
+      const Eigen::Quaterniond q(transform.rotation());
+      return Rotation{q.x(), q.y(), q.z(), q.w()};
+    }
+  };
+
+  /**
+   * kForever Sentinel value representing forever in time for static (unchanging) transforms
+   */
+  static constexpr std::chrono::milliseconds kForever = std::chrono::milliseconds::max();
+  static constexpr std::size_t kMaxTransformLengthDefault = 100U;
+
+  Transforms(std::size_t max_transform_length = kMaxTransformLengthDefault)
+      : max_transform_length_{max_transform_length} {}
+
+  /**
+   * UpdateTransform update a transform associated with the current time
+   *
+   * @param from the starting reference frame for the transform
+   * @param to the ending reference frame for the transform
+   * @param transform the actual transformation in terms of a translation and a rotation
+   * @param validity_window how long the transform is valid for (static transforms should use kForever)
+   */
+  void UpdateTransform(const std::string& from, const std::string& to, const RigidTransform& transform,
+                       std::chrono::milliseconds validity_window = kForever);
+
+  /**
+   * UpdateTransform update a transform associated with the given time
+   *
+   * @param from the starting reference frame for the transform
+   * @param to the ending reference frame for the transform
+   * @param transform the actual transformation in terms of a translation and a rotation
+   * @param validity_window how long the transform is valid for (static transforms should use kForever)
```

Comment:
Since the effective validity range is actually `2 * validity_window` aka `[when - validity_window : when + validity_window]` should we clarify the comment?

Reply:
Per discussion, decided to remove the concept of validity window altogether.